### PR TITLE
Added `pkg_config_name` and `pkg_config_aliases` information

### PR DESCRIPTION
**Pull request by franramirez688**
Docs for: https://github.com/conan-io/conan/pull/10084

### CODE REVIEW

File: reference/conanfile/tools/gnu/pkgconfigdeps.rst [hunk 1]
```diff
@@ -94,3 +94,118 @@ flags and require relations.
...
+
+        # any code here
+
+        def package_info(self):
+            self.cpp_info.set_property("pkg_config_name", "openssl")
+            self.cpp_info.components["crypto"].set_property("pkg_config_name", "libcrypto")
+            self.cpp_info.components["ssl"].set_property("pkg_config_name", "libssl")
+
+
+Making a simple ``conanfile.py`` to consume it:
```
[the remark this answers is not in the document]
Comment (franramirez688) [reply reference/conanfile/tools/gnu/pkgconfigdeps.rst:129]: That was really great! Thanks!

File: reference/conanfile/attributes.rst [hunk 2]
```diff
@@ -1073,6 +1073,7 @@ Properties related to *pkg_config*:
 - **pkg_config_name** property sets the ``names`` property for *pkg_config* generator.
 - **pkg_config_custom_content** property supported by the *pkg_config* generator that will add user
   defined content to the *.pc* files created by this generator
+- **pkg_config_aliases** property sets some aliases of any package/component name for *pkg_config* generator. This property only accepts list-like Python objects.
```
Comment (lasote) [inline reference/conanfile/attributes.rst:1076]: Talk with @czoido about this `reference/conanfile/attributest.rst` page, he's refactoring this stuff

File: reference/conanfile/tools/gnu/pkgconfigdeps.rst [hunk 3]
```diff
@@ -94,3 +94,144 @@ flags and require relations.
...
+
+Names and aliases
+++++++++++++++++++
+
+Aliases are available since: `1.43.0 <https://github.com/conan-io/conan/releases>`_
+
+By default, the ``*.pc`` file names will follow the rules defined below:
+
+* For packages, the own package name, e.g., package ``zlib/1.2.11`` -> ``zlib.pc``.
+* For components, the package name as namespace plus the own component one, package ``openssl/3.0.0`` and component ``openssl::crypto`` -> ``openssl-crypto.pc``.
```
Comment (lasote) [inline reference/conanfile/tools/gnu/pkgconfigdeps.rst:107]:
```suggestion
* For components, the package name + hypen + component name. e.g, ``openssl/3.0.0`` with `self.cpp_info.components["crytpo"]` -> ``openssl-crypto.pc``.
```

Don't use `::` here, that is cmake jargon stuff.

**Review by memsharded — APPROVED** [review]

File: reference/conanfile/tools/gnu/pkgconfigdeps.rst [hunk 4]
```diff
@@ -94,3 +94,144 @@ flags and require relations.
...
+
+
+Names and aliases
+++++++++++++++++++
+
+Aliases are available since: `1.43.0 <https://github.com/conan-io/conan/releases>`_
+
+By default, the ``*.pc`` file names will follow the rules defined below:
+
+* For packages, the own package name, e.g., package ``zlib/1.2.11`` -> ``zlib.pc``.
```
Comment (lasote) [inline reference/conanfile/tools/gnu/pkgconfigdeps.rst:106]:
```suggestion
* For packages, it is used the package name, e.g., ``zlib/1.2.11`` -> ``zlib.pc``.
```

File: reference/conanfile/tools/gnu/pkgconfigdeps.rst [hunk 5]
```diff
@@ -94,3 +94,144 @@ flags and require relations.
...
+* For components, the package name as namespace plus the own component one, package ``openssl/3.0.0`` and component ``openssl::crypto`` -> ``openssl-crypto.pc``.
+
+Given that, it's also possible to change easily that behavior thanks to ``pkg_config_name`` and ``pkg_config_aliases`` properties.
```
Comment (lasote) [inline reference/conanfile/tools/gnu/pkgconfigdeps.rst:109]:
```suggestion
You can change that default behavior with ``pkg_config_name`` and ``pkg_config_aliases`` properties.
```

**Review by lasote — APPROVED** [review]

File: reference/conanfile/tools/gnu/pkgconfigdeps.rst [hunk 6]
```diff
@@ -94,3 +94,144 @@ flags and require relations.
...
 *<PKG-NAME>.pc* file will declare all the components of the package as requires while the rest of the fields will be empty, relying on
 the propagation of flags coming from the components *<[PKG-NAME]-[COMP-NAME]>.pc* files.
+
+
+Names and aliases
+++++++++++++++++++
+
+Aliases are available since: `1.43.0 <https://github.com/conan-io/conan/releases>`_
+
+By default, the ``*.pc`` file names will follow the rules defined below:
```
Comment (tapia) [inline reference/conanfile/tools/gnu/pkgconfigdeps.rst:104]:
```suggestion
By default, the ``*.pc`` files will be named following these rules:
```

File: reference/conanfile/tools/gnu/pkgconfigdeps.rst (hunk 4 above)
Comment (tapia) [inline reference/conanfile/tools/gnu/pkgconfigdeps.rst]:
```suggestion
* For packages, the package name itself, e.g. ``zlib/1.2.11`` -> ``zlib.pc``.
```

File: reference/conanfile/tools/gnu/pkgconfigdeps.rst (hunk 3 above)
Comment (tapia) [inline reference/conanfile/tools/gnu/pkgconfigdeps.rst]:
```suggestion
* For components, the package name as namespace, plus the component name. For example: package ``openssl/3.0.0`` and component ``openssl::crypto`` -> ``openssl-crypto.pc``.
```

File: reference/conanfile/tools/gnu/pkgconfigdeps.rst (hunk 5 above)
Comment (tapia) [inline reference/conanfile/tools/gnu/pkgconfigdeps.rst]:
```suggestion
It's also possible to change easily this behavior using the ``pkg_config_name`` and ``pkg_config_aliases`` properties. For instance, ``openssl/3.0.0``` recipe has these ``pkg_config_name`` properties already declared:
```

File: reference/conanfile/tools/gnu/pkgconfigdeps.rst [hunk 7]
```diff
@@ -94,3 +94,144 @@ flags and require relations.
...
+
+* For packages, the own package name, e.g., package ``zlib/1.2.11`` -> ``zlib.pc``.
+* For components, the package name as namespace plus the own component one, package ``openssl/3.0.0`` and component ``openssl::crypto`` -> ``openssl-crypto.pc``.
+
+Given that, it's also possible to change easily that behavior thanks to ``pkg_config_name`` and ``pkg_config_aliases`` properties.
+
+For instance, ``openssl/3.0.0``` recipe has these ``pkg_config_name`` properties already declared:
```
Comment (tapia) [inline reference/conanfile/tools/gnu/pkgconfigdeps.rst:111]:
```suggestion
```

File: reference/conanfile/tools/gnu/pkgconfigdeps.rst [hunk 8]
```diff
@@ -94,3 +94,144 @@ flags and require relations.
...
+    Description: Conan component: libcrypto
+    Version: 3.0.0
+    Libs: -L"${libdir1}" -lcrypto -Wl,-rpath,"${libdir1}" -F Frameworks
+    Cflags: -I"${includedir1}"
+    Requires: zlib
+
+
+Now, let's see how ``pkg_config_aliases`` property works step by step.
+
+Let's create our own ``myopenssl/1.0.0`` recipe and define several aliases like these ones:
```
Comment (tapia) [inline reference/conanfile/tools/gnu/pkgconfigdeps.rst:162]:
```suggestion
Let's create our own ``myopenssl/1.0.0`` recipe and define several aliases like the following:
```

File: reference/conanfile/tools/gnu/pkgconfigdeps.rst [hunk 9]
```diff
@@ -94,3 +94,144 @@ flags and require relations.
...
+        name = "myopenssl"
+        version = "1.0.0"
+
+        def package_info(self):
+            # Aliases
+            self.cpp_info.set_property("pkg_config_aliases", ["myopenssl_alias"])
+            self.cpp_info.components["mycrypto"].set_property("pkg_config_aliases", ["mycrypto", "crp"])
+            self.cpp_info.components["myssl"].set_property("pkg_config_aliases", ["myssl"])
+
+Then, creating the package locally with :command:`conan create .` and consuming it :command:`conan install myopenssl/1.0.0@ -g PkgConfigDeps`, the files created will be:
```
Comment (tapia) [inline reference/conanfile/tools/gnu/pkgconfigdeps.rst:178]:
```suggestion
Then, after creating the package locally with :command:`conan create .` and consuming it :command:`conan install myopenssl/1.0.0@ -g PkgConfigDeps`, the files created will be:
```

File: reference/conanfile/tools/gnu/pkgconfigdeps.rst [hunk 10]
```diff
@@ -94,3 +94,144 @@ flags and require relations.
...
+++++++++++++++++++
+
+Aliases are available since: `1.43.0 <https://github.com/conan-io/conan/releases>`_
+
+By default, the ``*.pc`` file names will follow the rules defined below:
+
+* For packages, it uses the package name, e.g., package ``zlib/1.2.11`` -> ``zlib.pc``.
+* For components, the package name + hyphen + component name, e.g., ``openssl/3.0.0`` with ``self.cpp_info.components["crytpo"]``  -> ``openssl-crypto.pc``.
+
+You can change that default behavior with ``pkg_config_name`` and ``pkg_config_aliases`` properties.
```
Comment (tapia) [inline reference/conanfile/tools/gnu/pkgconfigdeps.rst:109]:
```suggestion
You can change that default behavior with the ``pkg_config_name`` and ``pkg_config_aliases`` properties. For instance, ``openssl/3.0.0``` recipe has these ``pkg_config_name`` properties already declared:
```

File: reference/conanfile/tools/gnu/pkgconfigdeps.rst [hunk 11]
```diff
@@ -94,3 +94,144 @@ flags and require relations.
...
+            self.cpp_info.components["ssl"].set_property("pkg_config_name", "libssl")
+
+Run :command:`conan install openssl/3.0.0@ -g PkgConfigDeps` and check the ``*.pc`` files created:
+
+- libcrypto.pc
+- libssl.pc
+- openssl.pc
+- zlib.pc *(openssl requires zlib)*
+
+Their ``pkg_config_name`` properties are used as absolute names for the PC files:
```
Comment (tapia) [inline reference/conanfile/tools/gnu/pkgconfigdeps.rst:134]: Not sure what this sentence means. What's an absolute name for a file?

File: reference/conanfile/tools/gnu/pkgconfigdeps.rst [hunk 12]
```diff
@@ -94,3 +94,144 @@ flags and require relations.
...
+
+.. code-block:: text
+    :caption: mycrypto.pc
+
+    Name: mycrypto
+    Description: Alias mycrypto for myopenssl-mycrypto
+    Version: 1.0.0
+    Requires: myopenssl-mycrypto
+
+Also, it's possible to use both properties together:
```
Comment (tapia) [inline reference/conanfile/tools/gnu/pkgconfigdeps.rst:198]:
```suggestion
It's also possible to use both properties together:
```

File: reference/conanfile/tools/gnu/pkgconfigdeps.rst (hunk 7 above)
Comment (franramirez688) [reply reference/conanfile/tools/gnu/pkgconfigdeps.rst:111]: I don't get why you're removing this line

Comment (franramirez688) [reply reference/conanfile/tools/gnu/pkgconfigdeps.rst:111]: I just saw your comment below 😁

**Review by czoido — APPROVED** [review]

**Pull request merged**